### PR TITLE
Jormungandr: expose bss realtime availability in /journeys

### PR DESCRIPTION
**Pull request by rchoquet**
### Exposes the realtime bss availability in /journeys for departure & arrival places.

Not sure if it lacks some tests or if I put some pieces of code in the wrong place \o/, let me now !

### CODE REVIEW

File: source/jormungandr/jormungandr/interfaces/v1/Journeys.py [hunk 1]
```diff
@@ -376,9 +378,13 @@ def __init__(self):
         parser_get.add_argument("_night_bus_filter_max_factor", type=float)
         parser_get.add_argument("_min_car", type=int)
         parser_get.add_argument("_min_bike", type=int)
+        parser_get.add_argument("bss_stands", type=boolean, default=False, description="Show bss stands availability")
 
         self.method_decorators.append(complete_links(self))
 
+        if parser_get.parse_args()["bss_stands"]:
```
Comment (xlqian) [inline source/jormungandr/jormungandr/interfaces/v1/Journeys.py]: prefer using `if parser_get.parse_args().get("bss_stands")`

File: source/jormungandr/jormungandr/parking_space_availability/bss/bss_provider_manager.py [hunk 2]
```diff
@@ -49,6 +49,14 @@ def handle_places(self, places):
                 place['poi'] = self.handle_poi(place['poi'])
         return places
 
+    def handle_journeys(self, journeys):
+        # iterate through each 'from' and 'to' place for each section
+        for place in (s[from_to] for j in journeys for s in j['sections'] for from_to in ('from', 'to') if from_to in s):
```
Comment (xlqian) [inline source/jormungandr/jormungandr/parking_space_availability/bss/bss_provider_manager.py]: `for place in (s.get(from_to, {}) for j in journeys for s in j.get('sections', []) for from_to in ('from', 'to'))`

File: source/jormungandr/tests/bss_provider_tests.py [hunk 3]
```diff
@@ -367,3 +366,47 @@ def pois_without_stands_on_places_autocompletion_test(self):
...
+        with mock_bss_providers(pois_supported=supported_pois):
+            query = journey_basic_query + "&first_section_mode=bss&last_section_mode=bss"
+            response = self.query_region(query)
+            self.is_valid_journey_response(response, query)
+
+            journeys = get_not_null(response, 'journeys')
+
+            for place in self._get_from_to_places_of_journeys_sections(journeys):
+                if 'poi' in place:
+                    assert 'stands' not in place['poi']
```
Comment (xlqian) [inline source/jormungandr/tests/bss_provider_tests.py]: L387 and L388 can be replaced by `assert 'stands' not in place.get('poi', {})`

File: source/jormungandr/tests/bss_provider_tests.py [hunk 4]
```diff
@@ -367,3 +366,47 @@ def pois_without_stands_on_places_autocompletion_test(self):
...
+                        assert 'stands' in poi
+                        assert poi['stands']['available_places'] == 13
+                        assert poi['stands']['available_bikes'] == 3
+                        assert poi['stands']['total_stands'] == 16
+                    else:
+                        assert 'stands' not in poi
+
+    def _get_from_to_places_of_journeys_sections(self, journeys):
+        # utility that returns 'from' and 'to' places for each section of each given journey
+        return (s[from_to] for j in journeys for s in j['sections'] for from_to in ('from', 'to') if from_to in s)
```
Comment (xlqian) [inline source/jormungandr/tests/bss_provider_tests.py]: `for place in (s.get(from_to, {}) for j in journeys for s in j.get('sections', []) for from_to in ('from', 'to'))`

Comment (TeXitoi) [inline source/jormungandr/tests/bss_provider_tests.py]: You can mutualize that with bss_provider_manager.py. A free function in bss_provider_manager.py will do the job. You can even include the filter on poi.

**Review by TeXitoi — CHANGES_REQUESTED** [review]

**Comment by rchoquet** [comment]
Jenkins, test this please.

**Review by TeXitoi — APPROVED** [review]

**Review by xlqian — APPROVED** [review]

**Pull request merged**